### PR TITLE
Improve and simplify code in promcfg

### PR DESCRIPTION
**Pull request by metalmatze**
Just a bunch of random things I found while looking into that file.

### CODE REVIEW

**Comment by brancz** [comment]
this needs a `make generate`, otherwise lgtm

**Comment by metalmatze** [comment]
Runing `make generate` doesn't change anything for me locally. :confused:

**Comment by brancz** [comment]
huh .. that's curious .. could you try `./scripts/check-make-generate.sh`?

File: pkg/prometheus/promcfg.go [hunk 1]
```diff
@@ -610,7 +608,7 @@ func generateRemoteWriteConfig(version semver.Version, specs []v1.RemoteWriteSpe
 		}
 
 		if spec.WriteRelabelConfigs != nil {
-			relabelings := []yaml.MapSlice{}
+			var relabelings []yaml.MapSlice
```
Comment (brancz) [inline pkg/prometheus/promcfg.go]: Actually looking at this again, I remember why we did this, consistently everywhere. This way we make sure we actually have a slice whereas with `var ...`, we may have nil, which can cause curious situations. I prefer actually initializing the empty slice.

**Pull request closed without merging**